examples/hif2a/generate_star_map.py

```diff
@@ -0,0 +1,264 @@
+# Construct a star map for the fep-benchmark hif2a ligands
+
+from pathlib import Path
+import numpy as np
+from rdkit import Chem
+from rdkit.Chem import rdFMCS, Draw
+
+import matplotlib.pyplot as plt
+
+root = Path(__file__).parent.parent.parent
+
+# 0. Get force field
+from ff import Forcefield
+from ff.handlers.deserialize import deserialize_handlers
+
+path_to_ff = str(root.joinpath('ff/params/smirnoff_1_1_0_ccc.py'))
+with open(path_to_ff) as f:
+    ff_handlers = deserialize_handlers(f.read())
+forcefield = Forcefield(ff_handlers)
+
+# 1. Get ligands
+# TODO: possibly Git submodule to fep-benchmark inside datasets/, rather than copying fep-benchmark into datasets
+
+path_to_ligands = str(root.joinpath('datasets/fep-benchmark/hif2a/ligands.sdf'))
+
+# locations relative to example folder
+path_to_results = Path(__file__).parent
+path_to_transformations = str(path_to_results.joinpath('relative_transformations.pkl'))
+
+supplier = Chem.SDMolSupplier(path_to_ligands, removeHs=False)
+mols = []
+for mol in supplier:
+    mols.append(mol)
+
+# 2. Identify ligand subset that shares a common substructure
+
+# SMARTS courtesy of YTZ (Jan 26, 2021)
+eyeballed_smarts_pattern = '[*]~1~[*]~[*]~2~[*]~[*]~[*]~[*](~[#8]~[*]~3~[*]~[*]~[*]~[*]~[*]~3)~[*]~2~[*]~1'
+bicyclic_query_mol = Chem.MolFromSmarts(eyeballed_smarts_pattern)
+
+# filter matches
+mols_with_core_1 = [mol for mol in mols if mol.HasSubstructMatch(bicyclic_query_mol)]
+
+
+# TODO: save this image...
+# Draw.MolsToGridImage(mols_with_core_1, molsPerRow=5, subImgSize=(200,200))
+
+# 3. Identify the center or "hub" of the star map
+
+# 3.1. Define a custom maximum common substructure search method
+
+class CompareDist(rdFMCS.MCSAtomCompare):
+    """Custom atom comparison: use positions within generated conformer"""
+
+    def compare(self, p, mol1, atom1, mol2, atom2):
+        """Atoms match if within 0.5 Å
+
+        Signature from super method:
+        (MCSAtomCompareParameters)parameters, (Mol)mol1, (int)atom1, (Mol)mol2, (int)atom2) -> bool
+        """
+        x_i = mol1.GetConformer(0).GetPositions()[atom1]
+        x_j = mol2.GetConformer(0).GetPositions()[atom2]
+        return bool(np.linalg.norm(x_i - x_j) <= 0.5)  # must convert from np.bool_ to Python bool!
+
+
+def mcs_map(a, b):
+    """Find the MCS map of going from A to B"""
+    params = rdFMCS.MCSParameters()
+    params.AtomTyper = CompareDist()
```

Comment:
Note that later on we may want to build compound queries to improve the precision of the MCS.

``` python
mcs_params = rdFMCS.MCSParameters()
mcs_params.AtomTyper = CompareDist()
mcs_params.BondCompareParameters.CompleteRingsOnly = 1 
```

However, we will need to update the RDKit to do so due to a bug I'd filed last year (and has since been resolved)

https://github.com/rdkit/rdkit/issues/3635